runtime/compiler/trj9/env/VMJ9.cpp

```diff
@@ -1292,7 +1298,11 @@ uintptrj_t
 TR_J9VMBase::getReferenceFieldAt(uintptrj_t objectPointer, uintptrj_t fieldOffset)
    {
    TR_ASSERT(haveAccess(), "Must haveAccess in getReferenceFieldAt");
-   return getReferenceFieldAtAddress(objectPointer + sizeof(J9Object) + fieldOffset);
+   if (TR::Compiler->om.shouldGenerateReadBarriersForFieldLoads())
+      return (uintptrj_t)vmThread()->javaVM->javaVM->memoryManagerFunctions->j9gc_objaccess_mixedObjectReadObject(vmThread(),
+         (J9Object*)objectPointer, J9_OBJECT_HEADER_SIZE + fieldOffset, false);
+   else
```

Comment:
Not sure if it matters for performance, but if it does, you can guard the if/else with J9VM_GC_COMPRESSED_POINTERS, ie:

```
-   return getReferenceFieldAtAddress(objectPointer + sizeof(J9Object) + fieldOffset);
+#if defined(OMR_GC_CONCURRENT_SCAVENGER)
+   if (TR::Compiler->om.shouldGenerateReadBarriersForFieldLoads())
+      return (uintptrj_t)vmThread()->javaVM->javaVM->memoryManagerFunctions->j9gc_objaccess_mixedObjectReadObject(vmThread(),
+         (J9Object*)objectPointer, J9_OBJECT_HEADER_SIZE + fieldOffset, false);
+   else
+#endif
+      return getReferenceFieldAtAddress(objectPointer + sizeof(J9Object) + fieldOffset);
```

Reply:
Do you mean by guarding the if/else with `OMR_GC_CONCURRENT_SCAVENGER`? I don't think there is a performance issue because `getReferenceFieldAt` is not called very often. Plus it is used at compile time, it should at least not affect the run time performance and I highly doubt it will affect the compilation time.

Reply:
>Do you mean by guarding the if/else with OMR_GC_CONCURRENT_SCAVENGER? 

Yeah sorry, typo on my part. 

> I don't think there is a performance issue because getReferenceFieldAt is not called very often.

Ah then I suppose it doesn't matter.